tests/HttpBase.py

```diff
@@ -0,0 +1,32 @@
+import unittest
+import threading
+from PyWebRunner import WebTester
+from six.moves import SimpleHTTPServer
+try:
+    import SocketServer
+except ImportError:
+    import socketserver as SocketServer
+
+
+class HTTPTestServer(SocketServer.TCPServer):
+    allow_reuse_address = True
+
+
+class HttpBase(unittest.TestCase):
```

Comment:
I pulled out the basic Http server logic into a reusable base class since tests are going to use that quite often.